crates/hyperswitch_connectors/src/connectors/nomupay/transformers.rs

```diff
@@ -33,58 +31,273 @@ impl<T> From<(StringMinorUnit, T)> for NomupayRouterData<T> {
+    pub destination_id: String,
+    pub payment_reference: String,
+    pub amount: FloatMajorUnit,
+    pub currency_code: String,
+    pub purpose: String,
+    pub description: String,
+    pub internal_memo: String,
+    pub release_on: String,
+    pub expire_on: String,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+#[serde(rename_all = "camelCase")]
+pub struct FeesType {
+    #[serde(rename = "type")]
+    pub fees_type: String,
+    pub fees: FloatMajorUnit,
+    pub currency_code: Currency,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+#[serde(rename_all = "camelCase")]
+pub struct PayoutQuoteResponse {
+    pub source_id: String,
+    pub destination_currency_code: Currency,
+    pub amount: FloatMajorUnit,
+    pub source_currency_code: Currency,
+    pub include_fee: bool,
+    pub fees: Vec<FeesType>,
+    pub payment_reference: String,
+}
+#[derive(Serialize, Deserialize, Debug)]
+#[serde(rename_all = "camelCase")]
+pub struct CommitResponse {
+    pub id: String,
+    pub status: String,
+    pub created_on: String,
+    pub last_updated: String,
+    pub source_id: String,
+    pub destination_id: String,
+    pub payment_reference: String,
+    pub amount: FloatMajorUnit,
+    pub currency_code: Currency,
+    pub purpose: String,
+    pub description: String,
+    pub internal_memo: String,
+    pub release_on: String,
+    pub expire_on: String,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct NomupayMetadata {
+    pub private_key: String,
+}
+
+#[derive(Default, Debug, Serialize, Deserialize, PartialEq)]
+#[serde(rename_all = "camelCase")]
+pub struct NomupayError1 {
+    pub error_code: String,
+    pub error_description: String,
+}
+
+#[derive(Default, Debug, Serialize, Deserialize, PartialEq)]
+#[serde(rename_all = "camelCase")]
+pub struct NomupayErrorType1 {
+    pub error: NomupayError1,
+}
+
+#[derive(Default, Debug, Serialize, Deserialize, PartialEq)]
+#[serde(rename_all = "camelCase")]
+pub struct ValidationError {
+    pub field: String,
+    pub message: String,
+}
+
+#[derive(Default, Debug, Serialize, Deserialize, PartialEq)]
+#[serde(rename_all = "camelCase")]
+pub struct NomupayError2 {
+    pub error_code: String,
+    pub validation_errors: Vec<ValidationError>,
+}
+
+#[derive(Default, Debug, Serialize, Deserialize, PartialEq)]
+#[serde(rename_all = "camelCase")]
+pub struct NomupayErrorType2 {
+    pub error: NomupayError2,
+}
+
+#[derive(Default, Debug, Serialize, Deserialize, PartialEq)]
+#[serde(rename_all = "camelCase")]
+pub struct DetailsType {
+    pub loc: Vec<String>,
+    #[serde(rename = "type")]
+    pub error_type: String,
+}
+
+#[derive(Default, Debug, Serialize, Deserialize, PartialEq)]
+#[serde(rename_all = "camelCase")]
+pub struct NomupayErrorType3 {
+    pub status_code: i64,
+    pub detail: Vec<DetailsType>,
+}
+
+#[derive(Debug, Serialize, Deserialize, PartialEq)]
+#[serde(untagged)]
+pub enum NomupayErrorResponse {
+    NomupayErrorType1(NomupayErrorType1),
```

Comment:
Can we model it like this to avoid untagged enums?

```suggestion
#[derive(Default, Debug, Serialize, Deserialize, PartialEq)]
#[serde(rename_all = "camelCase")]
pub struct ValidationError {
    pub field: String,
    pub message: String,
}

#[derive(Debug, Serialize, Deserialize, PartialEq)]
pub struct NomupayInnerError {
    pub error_code: String,
    pub error_description: Option<String>,
    pub validation_errors: Option<Vec<ValidationError>>,
}

#[derive(Default, Debug, Serialize, Deserialize, PartialEq)]
#[serde(rename_all = "camelCase")]
pub struct DetailsType {
    pub loc: Vec<String>,
    #[serde(rename = "type")]
    pub error_type: String,
}

#[derive(Debug, Serialize, Deserialize, PartialEq)]
pub struct NomupayErrorResponse {
    pub error: NomupayInnerError,
    pub status_code: Option<i64>,
    pub detail: Option<Vec<DetailsType>>,
}
```